app/services/foreman/renderer/scope/macros/loaders.rb

```diff
@@ -45,6 +45,25 @@ module Loaders
               example "<% #{name}.each_record do |#{model.to_s.downcase}| %>
   <%= #{model.to_s.downcase}.id %>, <%= #{model.to_s.downcase}.#{model == User ? 'login' : 'name'} %>
 <% end %>", desc: "Prints #{model.to_s.downcase} id and #{model == User ? 'login' : 'name'} of each #{model.to_s.downcase}"
+              example "# Joins - for comparing and filtering data
+# Uses INNER JOIN in one query
+# It may give a duplicate record in a few cases, the solution for it is using `select` with it.
+Host.joins(:interfaces).where(\"nics.mac = ?\", \"<mac>\")
+# => SELECT \"hosts\".* FROM \"hosts\" INNER JOIN \"nics\" ON \"nics\".\"host_id\" = \"hosts\".\"id\" WHERE \"hosts\".\"type\" = 'Host::Managed' AND (nics.mac = '<mac>')
+
+# Preload - for loading associated data
+# Two separate queries, doesn't have the (n+1) issue
+# `where` cannot be used for associated data filtering
+Host.preload(:interfaces)
+# => SELECT \"hosts\".* FROM \"hosts\" WHERE \"hosts\".\"type\" = 'Host::Managed'
+# => SELECT \"nics\".* FROM \"nics\" WHERE \"nics\".\"host_id\" IN (...) ORDER BY \"nics\".\"identifier\" ASC
+
+# Includes - for loading, comparing and filtering data with associated data
+# Loads associated data with either a single query for simple cases (in terms of joining table)
+# or with two separate queries for other cases
+# `where` can be used for associated data filtering
+# Doesn't have the (n+1) issue
```

Comment:
What is this about n+1 issue ?

Reply:
I forgot to mention it also for `join` and `preload`. I guess we don't have to mention it.